### PR TITLE
network ext_authz filter: support include_tls_session option from http variant

### PR DESCRIPTION
**Pull request by AdamEAnderson**
Commit Message: network ext_authz filter: support include_tls_session option from http variant
Additional Description: the http ext_authz filter supports [including tls session details](https://github.com/envoyproxy/envoy/blob/bd81cea070e5dbf45a60f56fa599822f8cef8ae1/api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto#L209-L213) in the check request. This PR ports that functionality over to the network ext_authz filter using the exact same mechanisms and the same field in the check request.
Risk Level: low
Testing: unit tests included
Docs Changes: docs are included in the protodoc for api changes. Slightly altered the protodoc for the check request attribute to link to both filters' config.
Release Notes: included as a new feature
Platform Specific Features: n/a

### CODE REVIEW

**Comment by AdamEAnderson** [comment]
CI currently failing due to flakes/failures not related to this PR. I will merge from main and rerun everything once I have a review to get things working again.

**Review by ggreenway — CHANGES_REQUESTED** [review]
Looks good overall. Thanks!

/wait

File: source/extensions/filters/common/ext_authz/check_request_utils.cc [hunk 1]
```diff
@@ -242,6 +242,15 @@ void CheckRequestUtils::createTcpCheck(
                      include_peer_certificate);
   setAttrContextPeer(*attrs->mutable_destination(), cb->connection(), server_name, true,
                      include_peer_certificate);
+  if (include_tls_session) {
+    if (cb->connection().ssl() != nullptr) {
```
[the remark this answers is not in the document]
Comment (AdamEAnderson) [reply source/extensions/filters/common/ext_authz/check_request_utils.cc]: yep, that makes sense

**Review by ggreenway — APPROVED** [review]
LGTM

**Comment by ggreenway** [comment]
> > Can you please merge main? There was a build break yesterday that you need the fix for.
> 
> @ggreenway merged main and most stuff works, but we're failing per-extension code coverage for extensions I haven't touched

Looking into it; when it's fixed, you'll need to merge main one more time.

**Comment by ggreenway** [comment]
The coverage issue should be fixed in main now; please merge main once more and hopefully CI will now pass.

/wait

**Comment by marc-barry** [comment]
I created https://github.com/envoyproxy/go-control-plane/issues/936 to have this added to the `go-control-plane`.

**Pull request merged**